### ISSUE TITLE
Artifacts from other workspace appear on current workspace

### ISSUE THREAD

**Issue by marko-avlijas** [linuxmint/Cinnamon#6253]
I can't reproduce this 100% of time, only about 10% of time or less.

Here is what is looks like: 
https://snag.gy/8DMo4Q.jpg

How to reproduce? Have 5 workspaces open, on workspace 2 have sublime text 3 opened.
Hover over a class or function definition so that mini window opens.
Then switch to workspace 1 using keyboard shortcut CTRL+ALT+LEFT Arrow

I have two monitors if that makes any difference, artifacts are always on same monitor.

Artifacts are above the normal windows like shown here:
https://snag.gy/SycPOu.jpg

**Comment by JosephMcc** [comment]
I can reproduce this. I've never seen it happen outside of sublime though so I'm not really sure where the issue actually is.

**Comment by marko-avlijas** [comment]
IMO no app should be able to do this, no matter how it's written. If there's an issue anywhere, it's not in the app, but in the desktop that allows apps to do this.

**Comment by JosephMcc** [comment]
I agree but it's gets more complicated than that. In most cases those little popup windows like sublime is using is technically it's own "window". Not sure what sort of requests it's sending. Is it asking to be "Sticky" or "Kept on top"? What type of window is it? Since it isn't open source we can't just look at the code and see what it's doing.

**Comment by marko-avlijas** [comment]
I have found another app that causes same bug - VLC media player.

    2.2.2 Weatherwax
    buildd@lcy01-06.buildd Feb 21 2017 23:23:51

Steps to reproduce:
1. open some video in vlc in fullscreen (playback paused)
2. hover mouse cursor near the bottom so that controls area is displayed
3. leave mouse cursor over controls area
4. switch workspace using CTRL+ALT+LEFT (or right)

VLC controls area is displayed on new workspace for like 5 seconds.
